azkaban-common/src/main/java/azkaban/project/NodeBean.java

```diff
@@ -17,18 +17,20 @@
 
 package azkaban.project;
 
+import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
 /**
- * The node bean is used by the YAML loader to deserialize DAG nodes
+ * Used by the YAML loader to deserialize DAG nodes in the flow
  */
-public class NodeBean {
+public class NodeBean implements Serializable {
 
   private String name;
   private Map<String, String> config;
   private List<String> dependsOn;
   private String type;
+  private List<NodeBean> nodes;
```

Comment:
maybe a more specific name?

Reply:
This name has to match what is defined in the flow YAML file. Since a node can either be a job or a sub-flow, I think it would be ok to have a generic name here.